src/Compiler/FileNameAndLineNumberAddingPreCompiler.php

```diff
@@ -27,12 +27,20 @@ public function __construct(
     ) {
     }
 
-    public function compileString(string $value): string
+    public function setFileNameAndCompileString(string $fileName, string $value): string
     {
-        if ($this->fileName === '') {
-            return '';
+        foreach ($this->configuration->getTemplatePaths() as $templatePath) {
+            $templatePath = rtrim($templatePath, '/') . '/';
+
+            if (str_contains($fileName, $templatePath)) {
+                $fileName = Str::after($fileName, $templatePath);
+                break;
+            }
         }
 
+        $this->fileName = $fileName;
```

Comment:
Great job!

Could you also remove this property to make service state-less?

Reply:
Some day I will be a _Developer_.